components/articles/Diary.js

```diff
@@ -1,41 +1,162 @@
-import React from 'react';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import moment from 'moment';
+import classNames from 'classnames';
 
-import { DiaryModel } from 'utils/customPropTypes';
+import Modal from 'components/common/Modal';
+import Text, { localize } from 'components/common/Text';
+import Clickable from 'components/common/Clickable';
 
-const LEFT_ARROW_KEY = 37;
-const RIGHT_ARROW_KEY = 38;
+import { actions as diaryActions, selectors as diarySelectors } from 'redux/ducks/diary';
+import { isToday } from 'utils/validators';
+import { DEFAULT_LOCALE } from 'constants';
 
-const Diary = ({ text, author, colloquialDate, getNextDiary, getPrevDiary }) => (
-  <div className="diary-article tile is-parent">
-    <article className="card tile is-child notification">
-      <h1 className="title">Дзённік дня</h1>
+const mapStateToProps = state => ({
+  diary: diarySelectors.getCurrent(state),
+});
+
+const mapDispatchToProps = {
+  getByDay: diaryActions.getByDay,
+  getNext: () => diaryActions.getClosest('next'),
+  getPrev: () => diaryActions.getClosest('prev'),
+};
+
+const NAV_BUTTONS = {
+  next: {
+    className: 'arrow-right',
+    iconClassName: 'fa-chevron-right',
+    title: 'diary.next',
+  },
+  prev: {
+    className: 'arrow-left',
+    iconClassName: 'fa-chevron-left',
+    title: 'diary.previous',
+  },
+};
+
+class Diary extends Component {
+  constructor() {
+    super();
+    this.state = { isModalActive: false };
+  }
+
+  componentDidMount() {
+    this.props.getByDay();
+  }
+
+  toggleModal = () => this.setState(prevState => ({ isModalActive: !prevState.isModalActive }));
+
+  renderDateElement = () => {
+    const { lang, diary: { date } } = this.props;
+    return (
       <div className="date">
-        <span>{colloquialDate}</span>
+        <div className="is-pulled-right">
+          {moment(date)
+            .locale(lang)
+            .format('LL')}
```

Comment:
please extract it to some constant, so we can use one date format across all app